indexer/services/ender/__tests__/lib/kafka-publisher.test.ts

```diff
@@ -139,11 +170,238 @@ describe('kafka-publisher', () => {
+          generateOrderSubaccountMessage(order, 'BTC-USD'),
+        ],
+      };
+      const message1: AnnotatedSubaccountMessage = {
+        blockHeight: '1',
+        transactionIndex: 1,
+        eventIndex: 1,
+        contents: JSON.stringify(msg1Contents),
+        subaccountId: {
+          owner: 'owner1',
+          number: 0,
+        },
+        version: '1',
+        orderId: 'order1',
+        isFill: true,
+        subaccountMessageContents: msg1Contents,
+      };
+
+      const msg2Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage(fill, 'ETH-USD'),
+        ],
+      };
+      const message2: AnnotatedSubaccountMessage = {
+        ...message1,
+        transactionIndex: 2,
+        contents: JSON.stringify(msg2Contents),
+        orderId: 'order2',
+        subaccountMessageContents: msg2Contents,
+      };
+
+      const msg3Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage({
+            ...fill,
+            size: '100',
+          }, 'BTC-USD'),
+        ],
+        orders: [
+          generateOrderSubaccountMessage({
+            ...order,
+            status: OrderStatus.FILLED,
+          }, 'BTC-USD'),
+        ],
+      };
+      const message3: AnnotatedSubaccountMessage = {
+        ...message1,
+        transactionIndex: 3,
+        contents: JSON.stringify(msg3Contents),
+        subaccountMessageContents: msg3Contents,
+      };
+
+      // non-fill subaccount message.
+      const msg4Contents: SubaccountMessageContents = generateTransferContents(
+        deposit,
+        testConstants.defaultAsset,
+        recipientSubaccountId,
+        undefined,
+        recipientSubaccountId,
+      );
+      const message4: AnnotatedSubaccountMessage = {
+        ...message1,
+        eventIndex: 4,
+        orderId: undefined,
+        isFill: undefined,
+        contents: JSON.stringify(msg4Contents),
+      };
+
+      const expectedMergedContents: SubaccountMessageContents = {
+        fills: [
+          msg1Contents.fills![0],
+          msg3Contents.fills![0],
+        ],
+        orders: [
+          msg3Contents.orders![0],
+        ],
+      };
+      const mergedMessage3: AnnotatedSubaccountMessage = {
+        ...message3,
+        contents: JSON.stringify(expectedMergedContents),
+        subaccountMessageContents: expectedMergedContents,
+      };
+
+      publisher.addEvents([
+        createConsolidatedKafkaEventFromSubaccount(message1),
+        createConsolidatedKafkaEventFromSubaccount(message2),
+        createConsolidatedKafkaEventFromSubaccount(message3),
+        createConsolidatedKafkaEventFromSubaccount(message4),
+      ]);
+
+      publisher.aggregateFillEventsForSubaccountMessages();
+      const expectedMsgs: SubaccountMessage[] = [
+        convertToSubaccountMessage(message4),
+        convertToSubaccountMessage(message2),
+        convertToSubaccountMessage(mergedMessage3),
```

Comment:
is this the right ordering? Shouldn't message 4 come after message 3?

Reply:
yes.

message 4 has txn index 1/event index 4.
msg 2 has txn index 2/ event index1.
msg 3 has txn index 3/event index 1.